### ISSUE TITLE
Fix Bug that makes it have to click twice

### ISSUE THREAD

**Issue by Yahid1** [uprm-inso4116-2024-2025-s1/semester-project-trolley-tracker-app#77]
There's bug that you have click twice to change tabs inside the DashBoard. This is unintended and needs to get fixed.

**Comment by JoelTorres20** [comment]
The bug was in the _auth.service.ts_ file, specifically in its constructor. Basically whenever a tab switch would occur and the constructor was called, a call to the method _handleLoginCallback()_ was being made as well. This would then re-route the app to the default state the first time a tab switch would occur. 

I fixed it by changing the call to the method _handleLoginCallback()_ to the _app.component.ts_ file, which is only supposed to run once per instance of the app. 

I also moved some console.logs that were inside the _get token()_ method, which is called a lot throughout the lifetime of the app, which were over-crowding the console with a lot of instances of the same token of the user.

Here are screenshots of the changes I made:

![Image](https://github.com/user-attachments/assets/894fd2dc-a2ae-4c6a-8e9b-2c21db505fc5)

--------------------------------------------------------------------------------------------------------

![Image](https://github.com/user-attachments/assets/05f58c31-6608-4208-8b3f-da4c76e3b335)
